frontends/web/src/routes/account/summary/chart.tsx

```diff
@@ -231,59 +286,57 @@ class Chart extends Component<Props, State> {
         lineColor: 'rgba(94, 148, 192, 1)',
         crosshairMarkerRadius: 6,
       });
-      switch (this.context.chartDisplay) {
+      switch (chartDisplay) {
       case 'week':
-        this.displayWeek();
+        displayWeek();
         break;
       case 'month':
-        this.displayMonth();
+        displayMonth();
         break;
       case 'year':
-        this.displayYear();
+        displayYear();
         break;
       }
-      const isChartDisplayWeekly = this.context.chartDisplay === 'week';
-      this.lineSeries.setData(
+      const isChartDisplayWeekly = chartDisplay === 'week';
+      lineSeries.current.setData(
         (isChartDisplayWeekly ?
-          this.props.data.chartDataHourly :
-          this.props.data.chartDataDaily) as ChartData
+          data.chartDataHourly :
+          data.chartDataDaily) as ChartData
       );
-      this.setFormattedData(
+      setFormattedData(
         (isChartDisplayWeekly ?
-          this.props.data.chartDataHourly :
-          this.props.data.chartDataDaily) as ChartData
+          data.chartDataHourly :
+          data.chartDataDaily) as ChartData
       );
-      this.chart.timeScale().subscribeVisibleLogicalRangeChange(this.calculateChange);
-      this.chart.subscribeCrosshairMove(this.handleCrosshair);
-      this.chart.timeScale().fitContent();
-      window.addEventListener('resize', this.onResize);
-      this.ref.current?.classList.remove(styles.invisible);
+      chart.current.timeScale().subscribeVisibleLogicalRangeChange(calculateChange);
+      chart.current.subscribeCrosshairMove(handleCrosshair);
+      chart.current.timeScale().fitContent();
+      window.addEventListener('resize', onResize);
+      ref.current?.classList.remove(styles.invisible);
     }
-  };
+  }
 
-  private reinitializeChart = () => {
-    this.removeChart();
-    this.createChart();
-  };
+  function reinitializeChart() {
+    removeChart();
+    initChart();
+  }
 
-  private removeChart = () => {
-    if (this.chart) {
-      this.chart.remove();
-      this.chart = undefined;
-      window.removeEventListener('resize', this.onResize);
+  function removeChart() {
+    if (chart.current) {
+      chart.current.remove();
+      chart.current = undefined;
+      window.removeEventListener('resize', onResize);
     }
-  };
+  }
 
-  private onResize = () => {
-    const isMobile = window.innerWidth <= 640;
-    this.setState({ isMobile });
```

Comment:
because it is calculated every time the component renders now. see [here](https://github.com/BitBoxSwiss/bitbox-wallet-app/pull/2728/files#diff-c761d5505dba4edd4496dfe892f927231782977bae81a155d645a6a04dd7ab59R88).

edit: ah, the component does not re-render automatically when the window re-sizes..good catch. I will convert `isMobile` to state again, and initialize it just like right now, but also `setIsMobile` in `onResize` then.

Reply:
can you past full links instead of "here", in some other comment you have like 3 links "here" "here" "this" this is hard for me to read.

also this should not update on ever render but on resize.